src/extends/room/intel.js

```diff
@@ -39,21 +40,54 @@ Room.prototype.saveIntel = function (refresh = false) {
 
   roominfo[INTEL_UPDATED] = Game.time
 
-  // Record room owner, level.
+  // Record room owner, level, and blocked exits
   if (this.controller) {
     if (this.controller.owner) {
       roominfo[INTEL_OWNER] = this.controller.owner.username
-      roominfo[INTEL_PRACTICAL_LEVEL] = this.getPracticalRoomLevel()
     } else if (this.controller.reservation) {
       roominfo[INTEL_OWNER] = this.controller.reservation.username
     } else if (roominfo[INTEL_OWNER]) {
       delete roominfo[INTEL_OWNER]
-      delete roominfo[INTEL_PRACTICAL_LEVEL]
     }
     if (this.controller.level) {
       roominfo[[INTEL_LEVEL]] = this.controller.level
+      roominfo[INTEL_PRACTICAL_LEVEL] = this.getPracticalRoomLevel()
     } else if (roominfo[INTEL_LEVEL]) {
       delete roominfo[INTEL_LEVEL]
+      delete roominfo[INTEL_PRACTICAL_LEVEL]
+    }
+
+    let candidates = this.find(FIND_SOURCES)
+    candidates.push(this.controller)
+    const centerish = this.getPositionAt(25, 25).findClosestByRange(candidates).pos
+    const exits = _.values(Game.map.describeExits(this.name))
+    const name = this.name
+    let blocked = []
+    for (let exit of exits) {
+      const targetPos = new RoomPosition(25, 25, exit)
+      const path = PathFinder.search(centerish, {pos: targetPos, range: 24},
+        {
+          swampCost: 1,
+          maxRooms: 2,
```

Comment:
Isn't it easier/better to just pathfind to the exits?

Reply:
If you want to move to an exit you have to first define the exits. This would mean-

1. Identify the exit id of the exit (right now we ignore that and only care about the room),
2. `find` all of the exit sources, which will us up some additional CPU,
3. *Either* add each exit source as a target for the pathfinder (so now we're building a new array of object, giving us another chance of triggering GC) or use something like findClosestByRange to limit things (potentially returning exits we can't each when others can be reached).

All of that would have to be done and it wouldn't change the results, just use more CPU. Since PathFinder uses a single map (rather than treating each room as a separate map) this method should work without pathfinder having to consider exit squares as a special case.

Reply:
Fair enough.